File: dispatch/java/runtime/adamma/c4dhi/claid/RemoteFunction/RemoteFunctionRunnableResult.java
package adamma.c4dhi.claid.RemoteFunction;

class RemoteFunctionRunnableResult
{
    Object returnValue;
    RemoteFunctionExecutionStatus status;

    public RemoteFunctionRunnableResult(Object returnValue, RemoteFunctionExecutionStatus status)
    {
        this.status = status;
        this.returnValue = returnValue;
    }

    public static RemoteFunctionRunnableResult makeSuccessfulResult(Object returnValue)
    {
        return new RemoteFunctionRunnableResult(returnValue, RemoteFunctionStatus.STATUS_OK);
    }

    public static RemoteFunctionRunnableResult makeFailedResult(RemoteFunctionStatus status)
    {
        return new RemoteFunctionRunnableResult(null, status);
    }

    public RemoteFunctionExecutionStatus getStatus()
    {
        return this.status;
    }

    public Object getReturnValue()
    {
        return this.returnValue;
    }
}

File: dispatch/ios/CLAID/Sources/CLAID/RemoteFunction/RemoteFunctionRunnableResult.swift
//
//  RemoteFunctionStatus.swift
//  CLAID
//
//  Created by Patrick Langer on 23.02.2025.
//


struct RemoteFunctionRunnableResult<T> {
    private let returnValue: T?
    private let status: Claidservice_RemoteFunctionStatus
    
    init(returnValue: T?, status: Claidservice_RemoteFunctionStatus) {
        self.returnValue = returnValue
        self.status = status
    }
    
    static func makeSuccessfulResult(_ returnValue: T) -> RemoteFunctionRunnableResult<T> {
        return RemoteFunctionRunnableResult(returnValue: returnValue, status: .statusOk)
    }
    
    static func makeFailedResult(status: Claidservice_RemoteFunctionStatus) -> RemoteFunctionRunnableResult<T> {
        return RemoteFunctionRunnableResult(returnValue: nil, status: status)
    }
    
    func getStatus() -> Claidservice_RemoteFunctionStatus {
        return self.status
    }
    
    func getReturnValue() -> T? {
        return self.returnValue
    }
}
